### ISSUE TITLE
Add some testing support for the REPL

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/ergo#279]
The REPL isn't being tested systematically.

**Comment by elissavetv** [comment]
Hiya! is this still open?

**Comment by padmakant** [comment]
i will work on this?can u explain this

**Comment by beingOptimusP** [comment]
hey i would like to work on this issue, can u tell more about it?

**Comment by mttrbrts** [comment]
> hey i would like to work on this issue, can u tell more about it?

Of course! The Ergo REPL (also called `ergotop`) is a command line utility that allows a developer to use Ergo code interactive. See the REPL documentation for more details. https://docs.accordproject.org/docs/ergo-repl.html

Guidance for building the Ergo compiler and REPL locally are available here:
https://github.com/accordproject/ergo/blob/e297da1228b3342598300b5744fe1a34d52ae815/DEVELOPERS.md#build-the-ergo-compiler-and-repl 

This issue requires adding some automated unit tests for the REPL to ensure that the REPL behaviour matches the behaviour of the engine on a particular platform. For example, I would recommend that you start by verifying that all of the examples in https://github.com/accordproject/ergo/blob/master/examples/repl_examples.ergo give the expected output in https://github.com/accordproject/ergo/blob/master/examples/repl_examples.result

One of the challenges with this issue is choosing a framework for running test suites against `stdin` and verifying the responses for `stdout`.

I you'd like to work on this, please confirm and I will assign to you.

**Comment by beingOptimusP** [comment]
yeah sure, its pretty intersting to work on, i will try my best
